bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TableOmDataSource.java

```diff
@@ -80,7 +78,7 @@ public void serialize(XMLStreamWriter xmlStreamWriter) throws XMLStreamException
             }
             for (int i = 0; i < structFields.length; i++) {
                 BField internalStructField = structFields[i];
-                int type = getReferredType(internalStructField.getFieldType()).getTag();
+                int type = internalStructField.getFieldType().getReferredType().getTag();
```

Comment:
Since there can be recursice type refrences Should we alyways use TypeUtils.getReferredType()?